packages/core-fl/src/FantasyLandStream.ts

```diff
@@ -0,0 +1,58 @@
+import { Stream, Sink, Scheduler, Disposable } from '@most/types'
+import {
+  continueWith,
+  empty,
+  filter,
+  map,
+  ap,
+  now,
+  chain,
+  never
+} from '../../core'
+
+interface FunctorFantasyLand<A> {
+  ['fantasy-land/map']<B>(fn: (a: A) => B): FunctorFantasyLand<B>
+}
+
+export const fantasyLand = <A>(stream: Stream<A>): FantasyLandStream<A> =>
+  new FantasyLandStream(stream)
+
+export class FantasyLandStream<A> implements Stream<A>, FunctorFantasyLand<A> {
+  constructor(private readonly stream: Stream<A>) {}
+
+  run(sink: Sink<A>, scheduler: Scheduler): Disposable {
+    return this.stream.run(sink, scheduler)
+  }
+
+  ['fantasy-land/concat'](nextStream: Stream<A>): FantasyLandStream<A> {
```

Comment:
There's probably some discussion to be had on what a proper stream monoid should be. There might even be more than 1 valid monoid instance.

The way you've used `continueWith` here is nice.  It's like a simple `switch`.  I've come to believe that `merge` is also a pretty good semigroup.

Any thoughts or insights on which might be the better default semigroup instance?  Are there others?  See my other, related comment about `zero` as well.